### PR TITLE
refactor(inference): retire the dead CLI-logins add-provider group (#2394)

### PR DESCRIPTION
**Pull request by oxoxDev**
## Summary

The LLM/Providers page's **Add a provider** dialog rendered a "CLI logins" group that was permanently empty and unselectable — `CLI_LOGINS_REACHABLE` was pinned `false`, so the category was handed `[]` and showed "Not available on this host." A group named after exactly the thing an operator wants, greyed out, reads as "ACP is switched off here" — while `GET {scope}/harnesses` already reports `claude` and `codex` as `detected: true, runsHere: true` on a desktop build.

This retires the group rather than finishing it. There is nothing a company-level row could ever hold for a CLI login: no key, no endpoint, and — per the External Harnesses design — deliberately no stored "connected" flag, since that would be a second source of truth that could disagree with the CLI actually being installed. Binding is per teammate and already works today from an agent's Model tab, so the dialog links there instead.

First of four PRs implementing #2394's rollout. Independent of the rest, in both directions.

## API Or Behavior Changes

- The add-provider dialog offers two categories (Cloud, Local runtimes) instead of three. It gains a line pointing at the roster: "Claude Code and Codex are bound per teammate, on an agent's Model tab. **Open the roster**" → `#/company/agents`, which closes the dialog on the way out.
- One deletion covers both mount sites — the Providers page and the first-run wizard's self-managed connect step, which mounts the same dialog.
- `CLI_LOGINS_REACHABLE` and `CLI_LOGINS_UNAVAILABLE` are deleted. Both lost their only consumer with the group; a `false` flag nothing reads is the same second-source-of-truth shape the category was retired for. The reasoning moves onto `addOptions`' surviving `cli` field, which stays populated because the self-managed step resolves an already-connected CLI row's label through it.
- `Category`'s now-unused `emptyNote` prop removed.
- No backend behavior change. The `cli_login` refusal stays exactly as it was; only its comment changes, from implying a future delegated-credential fix to stating the refusal is permanent by design.
- `Category::Cli`, `CLI_LOGINS`, `ProviderRef::ClaudeCode` and the `has_category` routing guard are all untouched — still load-bearing.
- No copy string changed on either side, so `catalogue_tests_mirror.rs` passes unmodified.

## Tests

New: `frontend/test/unit/inference-add-cli-group-retired.test.ts` — asserts no `#inference-add-cli` control, no "CLI logins" or "Not available on this host." text in the dialog, and that the roster link carries `#/company/agents` and closes the dialog when taken. Verified this fails (2/2) against the pre-change sources before passing, rather than passing vacuously.

- [x] `npm run typecheck`
- [x] `npm run typecheck:unit`
- [x] `npm run typecheck:e2e`
- [x] `npx vitest run` over the new test plus `inference-add-dialog-close`, `inference-connect`, `setup-wizard-self-managed-step1` — 60 passed
- [x] `scripts/ci/assert-setup-inference-imports.sh`
- [x] `scripts/ci/assert-design-tokens.sh`
- [x] `cargo fmt -p opencompany-core --check`
- [x] `cargo check -p opencompany-core`
- [x] `cargo clippy -p opencompany-core --no-deps --all-targets -- -D warnings`
- [ ] `cargo test` — N/A, no Rust behavior changed (comment only)

Manual verification in the running console (dev server against a live host): the dialog shows Cloud and Local runtimes only, with the pointer line beneath; clicking **Open the roster** navigates to the Agents roster and closes the dialog.

## Documentation

No spec change needed here. `docs/spec/runtime/external-harnesses-ui.md` describes the External Harnesses settings surface, which this PR does not touch — it is updated in the PR that folds that page into a per-harness detail view.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Changes**
  - The provider setup dialog now shows cloud and local provider options.
  - CLI login is no longer available as an add-provider option.
  - Added guidance explaining that Claude Code and Codex connections are managed per teammate.
  - Added an “Open the roster” link for quickly accessing teammate connection settings.
  - Simplified the empty state message when all available provider options are already connected.

### CODE REVIEW

File: frontend/src/inference/AddProviderDialog.tsx [hunk 1]
```diff
@@ -95,16 +95,20 @@ export function AddProviderDialog({
...
-            id="cli"
-            label={COPY.groupCli}
-            placeholder={COPY.placeholderCli}
-            helper={COPY.helperCli}
-            // Rendered saying so rather than hidden. See `CLI_LOGINS_REACHABLE`.
-            options={CLI_LOGINS_REACHABLE ? options.cli : []}
-            emptyNote={CLI_LOGINS_UNAVAILABLE}
-            onChoose={onChoose}
-          />
+
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/src/inference/AddProviderDialog.tsx:98]: Acknowledged: the AddProviderDialog e2e coverage gap is fair to flag. However, this PR is scoped to retire the CLI-logins group; adding a new e2e test for the dialog itself expands scope beyond the card's boundaries and belongs as a follow-up feature request to the console's e2e suite, not in this PR. The change is validated by the unit test added here (the absence of the group in the rendered state) and by manual verification during development. Coverage of the full dialog surface (including harness hint rendering) is reasonable follow-up work for the e2e suite.
